redbot/core/events.py

```diff
@@ -308,6 +308,11 @@ async def on_command_error(ctx, error, unhandled_by_cog=False):
 
     @bot.event
     async def on_message(message):
+        locale = await bot._i18n_cache.get_locale(message.guild)
+        regional_format = await bot._i18n_cache.get_regional_format(message.guild)
+        set_contextual_locale(locale)
+        set_contextual_regional_format(regional_format)
```

Comment:
Let's use our own APIs here:
```suggestion
        await set_contextual_locales_from_guild(bot, message.guild)
```